rasa/nlu/utils/spacy_utils.py

```diff
@@ -1,46 +1,66 @@
+from __future__ import annotations
+
 import typing
 import logging
 from typing import Any, Dict, List, Optional, Text, Tuple
 
-import rasa.shared.utils.io
-import rasa.utils.train_utils
-from rasa.nlu.components import Component
-from rasa.nlu.model import InvalidModelError
-from rasa.nlu.config import RasaNLUModelConfig
+
+from rasa.engine.graph import ExecutionContext, GraphComponent
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.nlu.constants import DENSE_FEATURIZABLE_ATTRIBUTES, SPACY_DOCS
 from rasa.shared.nlu.training_data.message import Message
 from rasa.shared.nlu.training_data.training_data import TrainingData
+import rasa.nlu.utils._spacy_utils
+import rasa.utils.train_utils
+from rasa.nlu.model import InvalidModelError
 from rasa.shared.constants import DOCS_URL_COMPONENTS
-from rasa.nlu.constants import SPACY_DOCS, DENSE_FEATURIZABLE_ATTRIBUTES
 
 logger = logging.getLogger(__name__)
 
 if typing.TYPE_CHECKING:
     from spacy.language import Language  # noqa: F401
-    from spacy.tokens.doc import Doc
-    from rasa.nlu.model import Metadata
+    from spacy.tokens import Doc
 
+# TODO: This is a workaround around until we have all components migrated to
+# `GraphComponent`.
+SpacyNLP = rasa.nlu.utils._spacy_utils.SpacyNLP
 
-class SpacyNLP(Component):
-    """The core component that links spaCy to related components in the pipeline."""
 
-    defaults = {
-        # when retrieving word vectors, this will decide if the casing
-        # of the word is relevant. E.g. `hello` and `Hello` will
-        # retrieve the same vector, if set to `False`. For some
-        # applications and models it makes sense to differentiate
-        # between these two words, therefore setting this to `True`.
-        "case_sensitive": False,
-    }
+class SpacyModel:
+    """Wraps `SpacyModelProvider` output to make it fingerprintable."""
 
-    def __init__(
-        self, component_config: Dict[Text, Any] = None, nlp: "Language" = None
-    ) -> None:
+    def __init__(self, model: Language, model_name: Text) -> None:
+        """Initializing SpacyModel."""
+        self.model = model
+        self.model_name = model_name
+
+    def fingerprint(self) -> Text:
+        """Fingerprints the model path.
+
+        Use a static fingerprint as we assume this only changes if the model name
+        changes and want to avoid investigating the model in greater detail for now.
 
-        self.nlp = nlp
-        super().__init__(component_config)
+        Returns:
+            Fingerprint for model.
+        """
+        return str(self.model_name)
```

Comment:
What are the consequences of a fingerprint collision (i.e. with a completely unrelated component)?

Reply:
If you see how we calculate the fingerprint key: https://github.com/RasaHQ/rasa/blob/03a3024fcb6cccc24cc43d5f450426477dba950c/rasa/engine/training/fingerprinting.py#L20

The fingerprint key represents the running of a single node and incorporates the a dict of the nodes inputs, input node name to input value fingerprint. So I don't think there would be an issue because it always relates to a specific link between two nodes.